frontend/src/context/ThemeContext.js

```diff
@@ -0,0 +1,26 @@
+import { createContext, useState, useEffect } from "react";
+
+export const ThemeContext = createContext();
+
+const ThemeContextProvider = ({ children }) => {
+  const [theme, setTheme] = useState(
+    window.matchMedia("(prefers-color-scheme: dark)").matches,
+  );
+
+  useEffect(() => {
+    document
+      .querySelector("body")
+      .setAttribute("data-bs-theme", theme ? "dark" : "light");
+  }, [theme]);
+
+  const handleTheme = () => {
+    setTheme(!theme);
```

Comment:
I would also rename `handleTheme` to something like `toggleDarkTheme`? You could also use another variant of setting state: `setDarkMode(mode => !mode)` - this will ensure that it always toggles appropriately even if it is called multiple times in a single update batch - but that's not a big deal though